routes/index.js

```diff
@@ -1,55 +1,406 @@
+                    owner: req.user._id
+                }, function(err, playlists) {
+                    User.findOne({
+                        _id: req.user._id
+                    }, function(err, user) {
+                        console.log(user.library)
+                        res.render('song', {
+                            song: song,
+                            rightTranslation: rightTranslation,
+                            isTranslationExisted: isTranslationExisted,
+                            translations: translations,
+                            playlists: playlists,
+                            inLibrary: user.library
+                        })
+                    })
+
                 })
+            } else {
+                res.render('song', {
+                    song: song,
+                    rightTranslation: rightTranslation,
+                    isTranslationExisted: isTranslationExisted,
+                    translations: translations,
+                    playlists: [],
+                    inLibrary: []
+                })
+            }
+        })
+    })
+
+router.route('/:song_id/add-translation')
+    .all(function(req, res, next) {
+        song_id = req.params.song_id;
+        song = {}
+        Song.findById(song_id, function(err, s) {
+            song = s;
+            next();
+        })
+    })
+    .get(isLoggedIn, function(req, res) {
+        var temp = '';
+        song.lyric.forEach(function(s) {
+            temp += s + '\n'
+        })
+        var messages = req.flash('error')
+        res.render('addTranslation', {
+            song: song,
+            stringLyric: temp,
+            messages: messages,
+            hasErrors: messages.length > 0
+        })
+    })
+    .post(function(req, res) {
+        req.checkBody('title_t', 'Title is required').notEmpty()
+        req.checkBody('lyric_t', 'Lyric is required').notEmpty()
+        var messages = req.validationErrors()
+
+        if (messages) {
+            res.render('addTranslation', {
+                messages: messages
             })
         } else {
-            res.render('songlist', {
-                songs: songs,
-                inLibrary: []
-            })
+            var lang = req.body.lang_t
+            var stringArr_t = req.body.lyric_t.split(/\r?\n|\//)
+            Song.findOne({
+                    source: song.id,
+                    lang: lang
+                }, function(err, translation) {
+                    if (err) {
+                        res.status(400).send('error ' + err)
+                    }
+                    var newSong = new Song({
+                        title: req.body.title_t,
+                        author: song.author,
+                        year: song.year,
+                        lang: lang,
+                        contributor: req.user.username,
+                        copyright: req.body.copyright_t,
+                        lyric: stringArr_t.slice(0),
```

Comment:
What does this _t convention mean? Is it necessary? Also, use a more descriptive name than stringArr. At least lyrics or lines or something.